src/main/java/com/zaxxer/sansorm/internal/Introspected.java

```diff
@@ -232,12 +236,12 @@ public boolean hasSelfJoinColumn()
    /**
     * Check if the introspected class has a self-join column defined.
     *
-    * @param columnName the column name to check
+    * @param columnName The column name to check. Requires case sensitive match of name element or property name without delimiters.
     * @return true if the specified column is a self-join column
     */
    public boolean isSelfJoinColumn(String columnName)
    {
-      return selfJoinFCInfo.columnName.equals(columnName);
+      return selfJoinFCInfo.getCaseSensitiveColumnName().equals(columnName);
```

Comment:
It is immediately clear that the columnName argument has to be case sensitive without deliters.